client/pkg/eventbus/eventbus.go

```diff
@@ -480,6 +478,39 @@ func (w *busWriter) AppendOne(ctx context.Context, event *ce.Event, opts ...api.
 	return encoded, nil
 }
 
+func (w *busWriter) SyncAppendOneStream(ctx context.Context, event *ce.Event, opts ...api.WriteOption) (eid string, err error) {
+	_ctx, span := w.tracer.Start(ctx, "SyncAppendOneStream")
+	defer span.End()
+
+	var writeOpts *api.WriteOptions = w.opts
+	if len(opts) > 0 {
+		writeOpts = w.opts.Copy()
+		for _, opt := range opts {
+			opt(writeOpts)
+		}
+	}
+
+	// 1. pick a writer of eventlog
+	lw, err := w.pickWritableLog(_ctx, writeOpts)
+	if err != nil {
+		return "", err
+	}
+
+	// 2. append the event to the eventlog
+	off, err := lw.SyncAppendStream(_ctx, event)
+	if err != nil {
+		return "", err
+	}
+
+	// 3. generate event ID
```

Comment:
Why generate event ID? what's the purpose of it?

Reply:
Communicated and reserved